[jit] Preserve constant from ConcreteModuleType to ClassType

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29220 [quant][graphmode] Enable foldbn tests
* #29219 [jit] Expose class constant through `attr` and `setattr` in object
* #30787 [jit] Add unsafeRemoveConstant for ClassType
* #31012 [jit] Fix getConstant
* **#29218 [jit] Preserve constant from ConcreteModuleType to ClassType**

Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

jerryzh168 committed:
[jit] Preserve constant from ConcreteModuleType to ClassType

Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zdevito reviewed (commented):
With this change, constants on modules should now get looked up from the ClassType rather than from the constants_ in the ConcreteModuleType. Otherwise, this doesn't test this is doing the right thing. `toTypeInferredIValue` is not the right call here. The compiler uses `toSugaredValue` which is a different call. `toTypeInferredIValue` is only used for tracing, not for the script.

jerryzh168 commented:
Should `ClassType` have `SugaredValue` in them? `SugaredValue` seems to be an intermediate value when we compile the module, and `ClassType` is the result of compilation, that's why I'm converting `py::object` to `IValue` here.  Or do you think we should keep using `py::object`/`SugaredValue` in the beginning and later convert it to `IValue`?

suo commented:
Summarizing our discussion offline today:
- We should delete `ConcreteModuleType::findConstant`, anybody looking for a constant should be forced to go through `jitType`.
- We need to do some refactoring to clean up `py::object` -> `IValue` translation. We want the following primitives:
  - Infer the JIT type of a `py::object` (already done, `tryToInferType()`)
  - Convert `py::object` to an `IValue` of the specified type (already done, `toIValue()`)
  - Check that a given JIT type is valid to be turned into a constant (not done)
Then we need to refactor `toSugaredValue()`, `_get_valid_constant`, `ConcreteModuleType::addConstant`, and `ClassType::addConstant()` to use these primitives in a consistent way.

Side note: we should maybe start calling it `ClassConstant` instead of `Constant`. The reason for this is that "constant" in the TorchScript sense and "constant" in the JIT IR sense are related but not identical concepts (for example, a tensor *can* be an IR constant but *can't* be a `ClassConstant`). This may be overly confusing though: what do you think @zdevito?

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jamesr66a commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -1,8 +1,51 @@
+bool isSimpleConstant(const IValue& v) {
+  return v.isBool() || v.isDouble() || v.isInt() || \
+    v.isString() || v.isNone() || v.isDevice();
+}
```
Can we put this somewhere more central? Like in ivalue.h or even as a method on IValue

jerryzh168 replied:
this is used only in class constant right now, not sure if this is useful for others in general.

jamesr66a replied:
I'm just worried about discoverability: someone might come across the need for the same concept in the future but won't know to look in `concrete_module_type.cpp`. If it's a method on IValue then they could see it in the class declaration or in their autocomplete

This is a nit, though, so up to you

jerryzh168 replied:
I think we can move this to ivalue.h when a new use case comes up

jerryzh168 replied:
we'll remove this part and keep this logic in python

jamesr66a commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -27,6 +70,17 @@ ClassTypePtr ConcreteModuleTypeBuilder::createTypeFromThis() const {
     cls->addAttribute(name, type, isParameter);
   }
 
+  for (const auto& pr : constants_) {
```
What does `pr` mean?

jerryzh168 replied:
pair I think

jamesr66a reviewed (commented):
Any tests?

@suo is going to be the best one to determine if this fits within the assumptions of ConcreteModuleType. I'll ping him to look

jerryzh168 commented:
@jamesr66a test comes later in the stack after we add support for module as well

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -1,8 +1,51 @@
+    v.isString() || v.isNone() || v.isDevice();
+}
+
+IValue getValidConstantFromTuple(const std::string& attr_name,
```
This entire function is copy/pasted from python without the python version removed. There should be only one source of truth for valid constants, or there will be future bugs.

jerryzh168 replied:
yeah python version is not used anymore, I can remove it

jerryzh168 replied:
actually I think I'll keep the python one. we'll just be converting from the preprocessed constant from python at this point

zdevito commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -27,6 +70,17 @@ ClassTypePtr ConcreteModuleTypeBuilder::createTypeFromThis() const {
+    const auto& val = pr.second.v_;
+    auto match = tryToInferType(val);
+    if (!match.success()) {
+      TORCH_CHECK(false, "Can't infer type of ", py::str(val), "\n:", match.reason());
```
This is not a great error message. It doesn't describe why we need to infer the type of this value.

zdevito commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -1,8 +1,51 @@
+                 [attr_name](const IValue& v) -> IValue {
+                   return getValidConstant(attr_name, v);
+                 });
+  return c10::ivalue::Tuple::create(valid_list);
```
This logic is very weird -- it turns lists into tuples? Why is this happening at this point? This seems like a way of grandfathering in python conversion behavior into IValue conversions.  I feel like it would be more appropriate for lists to be converted to tuples in python before it gets to this stage.

jerryzh168 replied:
Removed. I'll keep this logic in python

zdevito reviewed (commented):
Comments inline. We need to make sure this code is not redundant with its python counterparts to avoid bugs, and we need to make sure the definition of what we allow to be a constant is simple, documented, and implemented in a single place.

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
tests are failing right now, will fix tmr

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

suo commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -210,15 +224,6 @@ void ConcreteModuleTypeBuilder::addFailedAttribute(
   failedAttributes_.emplace(std::move(name), std::move(failureReason));
 }
 
-c10::optional<py::object> ConcreteModuleType::findConstant(
```
This method implementation was removed but it's still in the header file, please remove there as well.

suo commented on torch/csrc/jit/script/concrete_module_type.cpp:
```diff
@@ -27,6 +29,18 @@ ClassTypePtr ConcreteModuleTypeBuilder::createTypeFromThis() const {
+    const auto& val = pr.second.v_;
+    auto match = tryToInferType(val);
+    if (!match.success()) {
+      TORCH_CHECK(false, "We need to infer the type of constant to convert the python value to IValue, but failed to infer type of ", py::str(val), "\n:", match.reason());
```
This should be a `TORCH_INTERNAL_ASSERT` instead of a `TORCH_CHECK`, because our system is responsible for validating the precondition (that python value is convertible to a constant value).

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/script/python_sugared_value.h:
```diff
@@ -19,6 +19,12 @@ inline std::shared_ptr<SugaredValue> toSimple(Value* v) {
   return std::make_shared<SimpleValue>(v);
 }
 
+// entry point for instantiating a SugaredValue from an IValue
```
This isn't used outside of python_sugared_value.cpp, so it shouldn't be exposed.

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 92ec219:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=92ec21951210cd21001a00f956deb55f3cd8af01).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3920137) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** "pytorch android gradle build" ([details](https://dr.pytorch.org/build-details.html?build_id=72976874))
```
sys	0m0.164s
++ docker run --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -t -d -w /var/lib/jenkins 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-92ec21951210cd21001a00f956deb55f3cd8af01-android-x86_32
+ export id_x86_32=41ae7591bf57600f62d59a4fd89c4637165e153d1f6b4004f3ff8910005a3e45
+ id_x86_32=41ae7591bf57600f62d59a4fd89c4637165e153d1f6b4004f3ff8910005a3e45
+ export 'COMMAND=((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ COMMAND='((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ echo '((echo' '"source' './workspace/env"' '&&' echo '"sudo' chown -R jenkins 'workspace")' '|' docker exec -u jenkins -i '"$id_x86_32"' 'bash)' '2>&1'
+ unbuffer bash ./command.sh
+ ts
+ docker pull 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-92ec21951210cd21001a00f956deb55f3cd8af01-android-arm-v7a
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-92ec21951210cd21001a00f956deb55f3cd8af01-android-arm-v7a not found
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[jit] Preserve constant from ConcreteModuleType to ClassType"


Summary:
We need to be able to access constant in module.

Test Plan:
tbd

Reviewers:
mvz
Subscribers:

Tasks:

Tags:

Differential Revision: [D18846847](https://our.internmc.facebook.com/intern/diff/D18846847)

[ghstack-poisoned]
